src/main/java/commands/ViewScheCommand.java

```diff
@@ -0,0 +1,165 @@
+package commands;
+
+import members.Member;
+import core.Ui;
+import tasks.Deadline;
+import tasks.Event;
+import tasks.Last;
+import tasks.Period;
+import tasks.ToDo;
+import tasks.Task;
+import utils.DukeException;
+import utils.Parser;
+import utils.Storage;
+
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Date;
+
+public class ViewScheCommand extends Command {
+    private String line;
+
+    /**
+     * This is a class for command SCHEDULE, which displays tasks that has DATETIME in chronological order.
+     * Optionally provided DATE allows filtering of tasks from that day only
+     *
+     * @param line the DATE to be filtered
+     */
+    public ViewScheCommand(String line) {
+        this.line = line.trim();
+    }
+
+    @Override
+    public void execute(ArrayList<Task> tasks, ArrayList<Member> members, Storage storage) throws DukeException {
+        ArrayList<Task> scheTasks = new ArrayList<Task>();
+        String output = "Here is your schedule in order";
+        if (line.length() > 0) {
+            Date date = Parser.parseDate(line);
+            output += " on " + line;
+            ArrayList<Task> temp = (ArrayList<Task>) tasks.clone();
+            temp = removeNoTimeTask(temp);
+            temp = filterByDate(temp, date);
+            scheTasks = sortByDate(temp);
+        } else {
+            ArrayList<Task> temp = (ArrayList<Task>) tasks.clone();
+            scheTasks = sortByDate(temp);
+
+        }
+        for (int i = 0; i < scheTasks.size(); i++) {
+            output += "\n" + scheTasks.get(i);
+        }
+        Ui.print(output);
+    }
+
+
+    /**
+     * Removes and return list of tasks by provided date
+     * @param toFilter original list of tasks
+     * @param date date of tasks to be filtered by
+     * @return List of tasks filtered by date
+     */
+    private ArrayList<Task> filterByDate(ArrayList<Task> toFilter, Date date) {
+        ArrayList<Integer> toDelete = new ArrayList<Integer>();
+        Calendar calAim = Calendar.getInstance();
+        Calendar calTest = Calendar.getInstance();
+        calAim.setTime(date);
+
+        //remove all on different dates
+        for (int i = 0; i < toFilter.size(); i++) {
+            if (toFilter.get(i).getClass().equals(Deadline.class)) {
```

Comment:
Code is similar here and can be simplified. Is getStart() very different from getTime()?